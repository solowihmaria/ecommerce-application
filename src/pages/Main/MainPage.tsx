import React from 'react';
import { Header } from '../../components/blocks/Header/Header';
import styles from './MainPage.module.scss';

export const MainPage = () => (
    <div data-testid="main-page-test-id">
        <Header />
        <main className={styles.main}>
            <div className={styles.titleWrapper}>
                <h1 className={styles.title}>Main Page Content</h1>
            </div>
        </main>
    </div>
);
