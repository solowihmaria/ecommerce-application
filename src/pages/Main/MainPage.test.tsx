import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MainPage } from './MainPage';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

describe('<MainPage/> component', () => {
    test('Render MainPage component', () => {
        render(
            <MemoryRouter>
                <MainPage />
            </MemoryRouter>
        );
        const mainPage = screen.getByTestId('main-page-test-id');
        expect(mainPage).toBeInTheDocument();
        expect(mainPage).toMatchSnapshot();
    });
});
