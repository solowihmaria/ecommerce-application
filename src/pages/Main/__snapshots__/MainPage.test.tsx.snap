// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage/> component Render MainPage component 1`] = `
<div
  class="mainPageContainer"
  data-testid="main-page-test-id"
>
  <nav
    class="header"
    data-testid="test-id-header"
  >
    <div>
      <a
        class="iconLink"
        data-discover="true"
        href="/main"
      >
        <svg
          data-testid="mock-svg"
        />
      </a>
    </div>
    <div
      class="menu"
    >
      <a
        data-discover="true"
        href="/login"
      >
        <button
          class="button ghost headerButton"
        >
          Sign In
        </button>
      </a>
      <a
        data-discover="true"
        href="/register"
      >
        <button
          class="button ghost headerButton"
        >
          Sign Up
        </button>
      </a>
    </div>
  </nav>
  <main
    class="main"
  >
    <div
      class="titleWrapper"
    >
      <h1
        class="title"
      >
        Evergreen
      </h1>
      <h3
        class="titleSecondary"
      >
        Find your perfect plant
      </h3>
    </div>
  </main>
</div>
`;
